ls1intum/Artemis
# `General`: Add course exercise overview grouped by due date (timeframe view)

### Checklist
#### General



#### Client
Calculations in parent component for the weekly view will be moved to the child component in a follow up

Tests will be added in a Follow Up.

### Description


Improves the UX of the exercise Overview

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Course with / without exercises

1. View the `Course Overview` (see the exercise List)
2. See that the exercises are ordered properly within the groups (by due date, if equal by name)
3. Test that exercises can be expanded / collapsed 
4. Verify that filters work
5. Verify that search works _(that the categories do not uncollapse is a known UX issue)_
6. Verify that 1 Category is uncollapsed in the `Timeframe view` but `Past` shall never be uncollapsed by default when loading the page

~~Something I did not check yet: the guided tour may be broken (open TODO)~~
Checked on ts1, no guided tour seems to exist there

### Review Progress






#### Performance Review

#### Code Review
#### Manual Tests

### Test Coverage
Will be fixed and added in a follow up / before merge if time left

### Screenshots


NEW Version
![image](https://github.com/ls1intum/Artemis/assets/63976129/6d436597-55bb-4bc3-b6c7-832883de1a2e)

old Version
![image](https://github.com/ls1intum/Artemis/assets/63976129/5c5934c1-d9aa-4383-9c13-5f279d150189)
